src/smt/set_defaults.cpp

```diff
@@ -687,6 +687,12 @@ void SetDefaults::setDefaultsPost(const LogicInfo& logic, Options& opts) const
     }
   }
 
+  // if Alethe proofs, print lambda applications in non-curried manner
```

Comment:
This should not be done here, as it would clash with combinations of using `proof-format=alethe` with other features (e.g. dumping benchmarks).

Instead you should apply this property locally where Alethe proofs are printed.  As an example, see what LFSC does for printing terms: https://github.com/cvc5/cvc5/blob/main/src/proof/lfsc/lfsc_print_channel.cpp#L87.